src/main/scala/io/iohk/ethereum/validators/SignedTransactionValidator.scala

```diff
@@ -38,7 +45,7 @@ object SignedTransactionValidator extends SignedTransactionValidator {
     * @param stx Transaction to validate
     * @return Either the validated transaction or TransactionSyntaxError if an error was detected
     */
-  private def checkSyntacticValidity(stx: SignedTransaction): Either[SignedTransactionError, SignedTransaction] = {
+  private def checkSyntacticValidity(stx: SignedTransaction): Either[SignedTransactionError, Unit] = {
```

Comment:
I see your point. I'll add a `TransactionValid` case object :+1:
As we were using very similar return types for the other validators, should we also change their return type? Maybe this should be done in another PR but it would be quite easy to incorporate here.